op-service/ctxinterrupt/signal-waiter.go

```diff
@@ -0,0 +1,52 @@
+package ctxinterrupt
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+// defaultSignals is a set of default interrupt signals.
+var defaultSignals = []os.Signal{
+	// Let's not catch SIGQUIT as it's expected to terminate with a stack trace in Go. os.Kill
+	// should not/cannot be caught on most systems.
+	os.Interrupt,
+	syscall.SIGTERM,
+}
+
+type signalWaiter struct {
+	incoming chan os.Signal
+}
+
+func newSignalWaiter() signalWaiter {
+	catcher := signalWaiter{
+		// I'm not sure buffering these would have the intended effect beyond 1 as signals are
+		// generally emitted on timers and can't be relied on to deliver more than once in quick
+		// succession. It's not clear what the intention is if there are multiple concurrent waiters
+		// and only a single signal arrives.
```

Comment:
By buffering multiple signals one could consume at a different non-immediate pace. And some applications might want to e.g. force-shutdown upon receiving 3 interrupts. That said, yes the previous choice of buffering them was a bit much, we don't strictly need it, but it doesn't hurt either.